guvnor-ala/guvnor-ala-services-backend/src/main/java/org/guvnor/ala/services/backend/impl/PipelineServiceBackendImpl.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.services.backend.impl;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import org.guvnor.ala.pipeline.Input;
+import org.guvnor.ala.pipeline.PipelineConfig;
+import org.guvnor.ala.services.exceptions.BusinessException;
+import org.guvnor.ala.services.api.PipelineService;
+import org.guvnor.ala.services.api.backend.PipelineServiceBackend;
+import org.guvnor.ala.services.api.itemlist.PipelineConfigsList;
+import org.jboss.errai.bus.server.annotations.Service;
+
+@Service
+@ApplicationScoped
+public class PipelineServiceBackendImpl implements PipelineServiceBackend {
```

Comment:
Please forgive me if I am missing something, but I don't see why the `PipelineServiceBackend` is necessary (and the same applies for other `*Backend` services).

As far as I can tell this type just provides an Errai RPC endpoint for methods on `PipelineService`. But `PipelineService` is a JAX-RS interface so you can use a `Caller<PipelineService>` to make RPC calls to it directly from client-code. Is there a reason for having this extra layer in between?